src/subset.rs

```diff
@@ -220,7 +227,9 @@ impl<N: NodeIdT + Rand> Subset<N> {
                 return Ok(step);
             }
         };
-        self.broadcast_results.insert(proposer_id.clone(), value);
+        self.broadcast_results
+            .entry(proposer_id.clone())
+            .or_insert(Some(value));
```

Comment:
We need to make sure that we return the `Contribution` output even if our node is somehow lagging and receives the agreement output before the broadcast itself. I.e. at this point we need to check whether we already have an agreement output, and if it's `true`, return the `Contribution`.

(Alternatively `process_binary_agreement` could be modified to return the output in that case, too—not sure whether there's a clean way to do that.)